actions/dependabot-auto-triage/src/index.ts

```diff
@@ -0,0 +1,260 @@
+      } catch (error) {
+        if (error instanceof RequestError) {
+          if (error.status === 403) {
+            console.error(
+              `Error: Permission denied when trying to dismiss alerts.`,
+            );
+            console.error(
+              `Make sure the GITHUB_TOKEN has 'security-events: write' permission.`,
+            );
+            process.exit(1);
+          } else {
+            console.error(`API Error ${error.status}: ${error.message}`);
+            process.exit(1);
+          }
+        } else {
+          console.error(
+            "Error dismissing alerts:",
+            error instanceof Error ? error.message : String(error),
+          );
+          process.exit(1);
+        }
+      }
+    } catch (error) {
+      if (error instanceof RequestError) {
+        if (error.status === 404) {
+          console.error(
+            `Error: Repository ${owner}/${repo} not found or Dependabot alerts are not enabled.`,
+          );
+          console.error(
+            "Make sure Dependabot alerts are enabled in the repository's Security settings.",
+          );
+          process.exit(1);
+        } else if (
+          error.status === 403 ||
+          error.message.includes("Resource not accessible by integration")
+        ) {
+          console.error(`
+ERROR: Cannot access Dependabot alerts.
+
+This error may occur for several reasons:
+1. Dependabot alerts might not be enabled for this repository
+   - Enable them at: https://github.com/${owner}/${repo}/settings/security_analysis
+   
+2. The token being used might not have sufficient permissions
+   - For GITHUB_TOKEN: Make sure the workflow has 'security-events: write' permission
+   - For fine-grained tokens: Make sure the 'Dependabot alerts: write' permission is enabled
+
+3. Organization policies might be restricting access to security features
+   - Check with your organization administrator
+
+If you're sure the token has the right permissions, try accessing the alerts manually at:
+https://github.com/${owner}/${repo}/security/dependabot
+          `);
+          process.exit(1);
+        } else {
+          console.error(`API Error ${error.status}: ${error.message}`);
+          process.exit(1);
+        }
+      } else {
+        console.error(
+          "Error:",
+          error instanceof Error ? error.message : String(error),
+        );
+        process.exit(1);
+      }
+    }
+  } catch (error) {
+    console.error(
+      "Error:",
+      error instanceof Error ? error.message : String(error),
+    );
+    process.exit(1);
+  }
+}
+
+async function fetchAllAlerts(
+  octokit: Octokit,
+  owner: string,
+  repo: string,
+  alertTypes: string[],
+) {
+  const alerts = [];
+  let page = 1;
+  let hasMorePages = true;
+
+  try {
+    while (hasMorePages) {
+      // Using the dependabot alerts endpoint directly
+      const response = await octokit.request(
```

Comment:
octokit has a `paginate` function that should doe all the paging for you 🙂

Reply:
Done!